File: typescript/template/StringDisplay.ts
import { AbstractDisplay } from './AbstractDisplay.ts';

export class StringDisplay extends AbstractDisplay {
  private text: string;

  constructor(text: string) {
    super();
    this.text = text;
  }

  public open = (): string => {
    return '"';
  };

  public print = (isFinal: boolean): string => {
    if (isFinal) return this.text;
    return `${this.text}, `;
  };

  public close = (): string => {
    return '"';
  };
}

File: go/template/stringdisplay/stringdisplay.go
package stringdisplay

import (
	"github.com/hitsuji-haneta/design-pattern/go/template/display"
)

type stringDisplay struct {
	*display.AbstractDisplay
	text string
}

func New(text string) *stringDisplay {
	sd := &stringDisplay{&display.AbstractDisplay{}, text}
	sd.Printer = sd
	return sd
}

func (sd *stringDisplay) Open() string {
	return "["
}

func (sd *stringDisplay) Print(flag bool) string {
	if flag {
		return sd.text
	}
	return sd.text + ", "
}

func (sd *stringDisplay) Close() string {
	return "]"
}
